apps/android/firefox.py

```
@@ -74,28 +83,145 @@ def start(self, url="about:blank"):
             }
         )
 
-        device.push(profile.profile, device_profile)
-        device.chmod(device_storage, recursive=True)
+        self.device.push(self.profile.profile, device_profile)
+        self.device.chmod(device_storage, recursive=True)
 
-        app_args = [
-            "-profile",
-            device_profile,
-            "--marionette",
-            "--es",
-            "env0",
-            "LOG_VERBOSE=1",
-            "--es",
-            "env1",
-            "R_LOG_LEVEL=6",
-        ]
+    def run_android_app(self, url):
+        pass
 
-        # start app
-        device.stop_application(self.APP_NAME)
-        device.launch_activity(
+
+class GeckoViewExampleApp(AndroidFirefox):
+    APP_NAME = "org.mozilla.geckoview_example"
+    ACTIVITY_NAME = "GeckoViewActivity"
+
+    def __init__(self, *args, **kwargs):
+        super(GeckoViewExampleApp, self).__init__(*args, **kwargs)
+
+    def run_android_app(self, url):
+        self.device.stop_application(self.APP_NAME)
+        self.device.launch_activity(
             self.APP_NAME,
             self.ACTIVITY_NAME,
-            extra_args=app_args,
+            extra_args=self.app_args,
             url=url,
             e10s=True,
-            fail_if_running=False,
+            fail_if_running=False
+        )
+
+    def start(self, url="about:blank"):
+        # create profile
+        self.set_profile()
+        # create certificate database
+        self.create_certificate()
+
+        # setup device
+        self.setup_device()
+
+        # start app
+        self.run_android_app(url)
+
+
+class FenixApp(AndroidFirefox):
+    APP_NAME = "org.mozilla.fenix.raptor"
+    ACTIVITY_NAME = "org.mozilla.fenix.browser.BrowserPerformanceTestActivity"
+    DEFAULT_INTENT = "android.intent.action.VIEW"
```

I think `DEFAULT` is misleading here, let's rename this to simply `INTENT`.